src/ng/directive/form.js

```diff
@@ -168,6 +168,7 @@ function FormController(element, attrs, $scope, $animate, $interpolate) {
     });
 
     arrayRemove(controls, control);
+    control.$$parentForm = nullFormCtrl;
```

Comment:
I think a similar change is need in [`$addControl()`][1]: `control.$$parentForm = form`

[1]: https://github.com/angular/angular.js/pull/12418/files#diff-1838ed9bd28307b6d42de98549826ae5R134